test cases/common/48 pkgconfig-gen/meson.build

```diff
@@ -37,7 +36,8 @@ test('pkgconfig-validation', pkgconfig,
 # Test that name_prefix='' and name='libfoo' results in '-lfoo'
 lib2 = shared_library('libfoo', 'simple.c',
   name_prefix : '',
-  version : libver)
+  version : libver,
+  include_directories : include_directories('.'))
```

Comment:
This is simply wrong. The current source and build directories are always guaranteed to be in the search path and our documentation explicitly says that you should not use `include_directories('.')`. This needs a different solution.

Reply:
Right, I changed to always add `-I` cflag to the library's subdir in both source and build trees implicitly.